bids/layout/index.py

```diff
@@ -37,62 +51,34 @@ def _check_path_matches_patterns(path, patterns, root=None):
 
     # Path now can be downcast to str
     path = str(path)
-    for patt in patterns:
-        if hasattr(patt, "search"):
-            if patt.search(path):
-                return True
-            else:
-                continue
-
-        if isinstance(patt, Path):
-            patt = str(
-                patt if root is None
-                else Path("/") / patt.relative_to(root)
-            )
-        if str(patt) in path:
-            return str(patt)
 
+    for patt in patterns:
+        if patt.search(path):
+            return True
     return False
 
 
-def _validate_path(path, incl_patt=None, excl_patt=None, root=None, default=None):
-    incl_matched = _check_path_matches_patterns(path, incl_patt, root=root)
-    excl_matched = _check_path_matches_patterns(path, excl_patt, root=root)
-
-    if incl_matched is False and excl_matched is False:
-        return default
-
-    # Include: if a inclusion regex pattern matched or a nonregex inclusion pattern matched
-    if incl_matched is True or (incl_matched and not excl_matched):
-        return True
-
-    if excl_matched is True:
-        return False
-
-    if not incl_matched and excl_matched:
-        return False
-
-    # Both matched: nested pattern
-    if incl_matched not in excl_matched:
+def _validate_path(path, incl_patt=None, excl_patt=None, root=None):
+    if _check_path_matches_patterns(path, incl_patt, root=root):
         return True
 
-    if excl_matched not in incl_matched:
+    if _check_path_matches_patterns(path, excl_patt, root=root):
         return False
 
-    return default
-
 
 class BIDSLayoutIndexer:
     """ Indexer class for BIDSLayout.
 
     Parameters
     ----------
-    validate : bool, optional
+    valid_only : bool, optional
         If True, all files are checked for BIDS compliance when first indexed,
         and non-compliant files are ignored. This provides a convenient way to
         restrict file indexing to only those files defined in the "core" BIDS
-        spec, as setting validate=True will lead files in supplementary folders
-        like derivatives/, code/, etc. to be ignored.
+        spec, as setting ``valid_only=True`` will lead noncompliant files
+        like ``sub-01/nonbidsfile.txt`` to be ignored.
+    validate : bool, optional
+        Deprecated. Please use ``valid_only``.
```

Comment:
Do we have to break the API in a fix? However badly the term may be named, you're not making a distinction in behavior between `validate` and `valid_only`. This is going to add noise to bug people to fix up their use of an API that we intend to remove soon anyway.

Reply:
This doesn't break compatibility. Also, the ``BIDSLayout``, which is what 80% of users need has not changed the API at all.

For maintainability, even if short lived, I think this change is justified: to me, as a spotty contributor, `validate` was confusing and made coding harder.

I'm fine rolling back to `validate`, but TBH, I don't see this particularly irritating while helps make the API more clear.

Reply:
I vote we don't break the API, even though I agree w/ the spirit of @oesteban 's changes. I think it makes sense to change in principle, but like @effigies said we are working (?) replacing pybids w/ ancpbids, so I rather push such changes to that major change.